src/main/java/com/ably/kafka/connect/ChannelSinkTask.java

```diff
@@ -98,30 +96,29 @@ public void start(Map<String, String> settings) {
 
         try {
             ably = new AblyRealtime(config.clientOptions);
-            channel = ably.channels.get(config.channelName, config.channelOptions);
         } catch(AblyException e) {
             logger.error("error initializing ably client", e);
         }
     }
 
     @Override
     public void put(Collection<SinkRecord> records) {
-        if (channel == null) {
+        if (ably == null) {
             // Put is not retryable, throwing error will indicate this
             throw new ConnectException("ably client is uninitialized");
         }
 
-        for (SinkRecord r : records) {
+        for (SinkRecord record : records) {
             // TODO: add configuration to change the event name
             try {
-                Message message = new Message("sink", r.value());
-                message.id = String.format("%d:%d:%d", r.topic().hashCode(), r.kafkaPartition(), r.kafkaOffset());
+                Message message = new Message("sink", record.value());
```

Comment:
I agree that it's weird  and I actually brought up in my comment here https://github.com/ably/kafka-connect-ably/pull/38#issuecomment-1048623343. 
We basically do not have any configuration for the message. We will need to decide how we want to configure our message so that I can either add a method to our current interface (or create a new one for message translation if we want to keep message translation separate from channel).

Reply:
Just appending this link to say we have an open issue for this https://github.com/ably/kafka-connect-ably/issues/41

Reply:
Resolving this conversation as there was no more interaction and this PR was approved by @jaley